jib-core/src/main/java/com/google/cloud/tools/jib/image/ReproducibleLayerBuilder.java

```diff
@@ -176,6 +176,6 @@ public Blob build() throws IOException {
       tarStreamBuilder.addTarArchiveEntry(entry);
     }
 
-    return Blobs.from(tarStreamBuilder::writeAsTarArchiveTo);
+    return Blobs.from(tarStreamBuilder::writeAsTarArchiveTo, true);
```

Comment:
Just noticed that generally speaking `TarStreamBuilder::writeAsTarArchiveTo` is not retryable, as non-retryable `Blob`s [can be added](https://github.com/GoogleContainerTools/jib/blob/1fd6b62c90bb96a53fdc719265c4224157810361/jib-core/src/main/java/com/google/cloud/tools/jib/tar/TarStreamBuilder.java#L92) to it.
```suggestion
    return Blobs.from(tarStreamBuilder::writeAsTarArchiveTo, false);
```

Reply:
It does reopen the stream  so should be retryable no?
Dont have it in mind but think it must be retryable somehow or we will get troubles later.

Reply:
The method `writeAsTarArchiveTo` is retryable if and only if all of the added blobs are retryable, which isn't guaranteed (even if it may be the case that no non-retryable blobs are currently added in our codebase). So I think `false` is the right thing to do.

We won't get into a trouble later because of this. The only place this is used is when [assembling an application layer](https://github.com/GoogleContainerTools/jib/blob/1fd6b62c90bb96a53fdc719265c4224157810361/jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/BuildAndCacheApplicationLayerStep.java#L108-L109), and we do cache and return a new blob from a cache file using `Blobs.from()`. That blob is retryable, so your retry logic will certainly work for pushing application layers. There is no need to insist `true` here only for the sake of making the logic work. `false` is conceptually correct, so let's do that.